File: tests/rosetta/transpiler/Java/determine-if-a-string-is-numeric-2.java
public class Main {

    static boolean isInt(String s) {
        if (_runeLen(s) == 0) {
            return false;
        }
        for (int _i = 0; _i < s.length(); _i++) {
            var ch = s.substring(_i, _i + 1);
            if ((ch.compareTo("0") < 0) || (ch.compareTo("9") > 0)) {
                return false;
            }
        }
        return true;
    }

    static void main() {
        System.out.println("Are these strings integers?");
        String v = "1";
        boolean b = false;
        if (isInt(v)) {
            b = true;
        }
        System.out.println("  " + v + " -> " + String.valueOf(b));
        String i = "one";
        System.out.println("  " + i + " -> " + String.valueOf(isInt(i)));
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            main();
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static int _runeLen(String s) {
        return s.codePointCount(0, s.length());
    }
}

File: tests/rosetta/transpiler/FS/determine-if-a-string-is-numeric-2.fs
// Generated 2025-07-31 00:10 +0700

exception Return

let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec isInt (s: string) =
    let mutable __ret : bool = Unchecked.defaultof<bool>
    let mutable s = s
    try
        if (String.length s) = 0 then
            __ret <- false
            raise Return
        for ch in s do
            if (ch < "0") || (ch > "9") then
                __ret <- false
                raise Return
        __ret <- true
        raise Return
        __ret
    with
        | Return -> __ret
and main () =
    let mutable __ret : unit = Unchecked.defaultof<unit>
    try
        let __bench_start = _now()
        let __mem_start = System.GC.GetTotalMemory(true)
        printfn "%s" "Are these strings integers?"
        let v: string = "1"
        let mutable b: bool = false
        if isInt v then
            b <- true
        printfn "%s" ((("  " + v) + " -> ") + (string b))
        let i: string = "one"
        printfn "%s" ((("  " + i) + " -> ") + (string (isInt i)))
        let __bench_end = _now()
        let __mem_end = System.GC.GetTotalMemory(true)
        printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

        __ret
    with
        | Return -> __ret
main()
